test/functional/signature/cert-chain.bats

```diff
@@ -0,0 +1,158 @@
+
+	key_generation_cmd=$(cat <<-EOM
+			# Create a root key and cert
+			openssl req -x509 -sha512 -days 1 -newkey rsa:4096  -keyout "$TEST_NAME/root.key" -out "$TEST_NAME/root.pem" -nodes -subj '/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=localhost'
+			# Create a intermediate key and cert signed by root
+			openssl req -sha512 -nodes -newkey rsa:4096 -keyout "$TEST_NAME/intermediate.key" -out "$TEST_NAME/intermediate.csr" -subj '/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=intermediate'
+			openssl x509 -days 1 -req -in "$TEST_NAME/intermediate.csr" -CA "$TEST_NAME/root.pem" -CAkey "$TEST_NAME/root.key" -set_serial 0x1 -out "$TEST_NAME/intermediate.pem"
+			# Create a self signed certificate for intermediate
+			openssl req -days 1 -new -x509 -key "$TEST_NAME/intermediate.key" -out "$TEST_NAME/intermediate_self_signed.pem" -set_serial 0x1 -subj '/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=intermediate'
+			# Sign the MoM with intermediate cert
+			openssl smime -sign -binary -in "$WEBDIR"/10/Manifest.MoM -signer "$TEST_NAME/intermediate.pem" -inkey "$TEST_NAME/intermediate.key" -out "$WEBDIR"/10/Manifest.MoM.sig -outform DER
+			# Sign the MoM with self signed intermediate cert
+			openssl smime -sign -binary -in "$WEBDIR"/10/Manifest.MoM -signer "$TEST_NAME/intermediate_self_signed.pem" -inkey "$TEST_NAME/intermediate.key" -out "$WEBDIR"/10/Manifest.MoM.sig_self_signed -outform DER
+			# Create a certificate store with both root and self signed intermediate
+			cat "$TEST_NAME/root.pem" > "$TEST_NAME/trust.pem"
+			cat "$TEST_NAME/intermediate_self_signed.pem" >> "$TEST_NAME/trust.pem"
+	EOM
+	)
+
+	run sudo sh -c "$key_generation_cmd"
+	assert_status_is 0
+
+}
+
+@test "SIG010: MoM signed with intermediate key should fail" {
+
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS test-bundle"
+
+	assert_status_is "$SWUPD_COULDNT_LOAD_MOM"
+	expected_output=$(cat <<-EOM
+		Error: FAILED TO VERIFY SIGNATURE OF Manifest.MoM version 10!!!
+		Error: Cannot load official manifest MoM for version 10
+	EOM
+	)
+	assert_in_output "$expected_output"
+	assert_file_not_exists "$TARGETDIR"/test-file
+
+}
+
+@test "SIG011: MoM signed with intermediate key and root cert is used" {
+
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS_NO_CERT -C $TEST_NAME/root.pem test-bundle"
+
+	assert_status_is 0
+	expected_output=$(cat <<-EOM
+		Loading required manifests...
+		No packs need to be downloaded
+		Starting download of remaining update content. This may take a while...
+		Installing bundle(s) files...
+		Calling post-update helper scripts
+		Successfully installed 1 bundle
+	EOM
+	)
+	assert_in_output "$expected_output"
+	assert_file_exists "$TARGETDIR"/test-file
+
+}
+
+@test "SIG012: MoM signed with self-signed cert, but using root to validate" {
+
+	# replace sig file to one signed with intermediate_self_signed.pem
+	run sudo sh -c "cp "$WEBDIR"/10/Manifest.MoM.sig_self_signed "$WEBDIR"/10/Manifest.MoM.sig"
```

Comment:
same comment from above about using "run" to execute a test prerequisite, if you use run and that command fails you are just gonna mask the error, and makes the test less straightforward since it makes you think you are testing that action.